src/build/buildExecutor.ts

```diff
@@ -65,6 +72,43 @@ export class BuildExecutor {
         return buildResult;
     }
 
+    public getLanguageClient(input: BuildInput, buildUserToken: string): LanguageClient {
+        let buildParameters = this.getBuildParameters(undefined, input, buildUserToken);
+        if (buildUserToken) {
+            buildParameters.envs["X-OP-BuildUserToken"] = buildUserToken;
+        }
+        let isPublic = !buildUserToken;

```

Comment:
when we already have a config to indicate the user role, we should not use this anymore, and there are several corner cases you may need to handle when we we support the user role:
1. If the user role is public contributor, you need to clean the credential stored in the local storage, or the statubar will still display the original credential information.
You can do those in s separated PR.